### PR TITLE
Feature/implement properties file configuration for java agent epmdj 10622

### PR DESCRIPTION
**Pull request by epamrsa**

### CODE REVIEW

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt [hunk 1]
```diff
@@ -57,6 +57,7 @@ object Agent {
         println(LOGO)
         try {
             defaultNativeLoggingConfiguration()
+            drillInstallationDir = getDrillInstallationDir()
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:60]: declare either `var` or `val` identifier (its never reassigned so `val` I guess)

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt [hunk 2]
```diff
@@ -87,11 +88,18 @@ object Agent {
 
     private val pathSeparator = if (Platform.osFamily == OsFamily.WINDOWS) "\\" else "/"
 
+    private fun getDrillInstallationDir() = open("/proc/self/cmdline", O_RDONLY).let(::Input).readText().run {
+        val isContainsAgetPath: (String) -> Boolean = { it.contains("-agentpath:") }
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:92]: fix the typo `isContainsAgetPath` - `isContainsAgentPath`

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt [hunk 3]
```diff
@@ -87,11 +88,18 @@ object Agent {
 
     private val pathSeparator = if (Platform.osFamily == OsFamily.WINDOWS) "\\" else "/"
 
+    private fun getDrillInstallationDir() = open("/proc/self/cmdline", O_RDONLY).let(::Input).readText().run {
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:91]: `"/proc/self/cmdline"`- is this crossplatform?

I have no idea if prev implementation ([JavaProcessInfo](https://github.com/Drill4J/java-agent/blob/main/java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/process/JavaProcessInfo.kt#L34C43-L34C43) and ["ps -p *pid* -o args"](https://github.com/Drill4J/lib-jvm-shared/blob/main/agent/src/posixMain/kotlin/com/epam/drill/agent/configuration/FileUtils.kt#L25) ) were crossplatform either

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt [hunk 4]
```diff
@@ -87,11 +88,18 @@ object Agent {
...
+        val isContainsAgetPath: (String) -> Boolean = { it.contains("-agentpath:") }
+        val agentLine = this.takeIf(isContainsAgetPath) ?: getenv("JAVA_TOOL_OPTIONS")!!.toKString()
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:93]: In this case `.run` makes method code harder to comprehend
Consider:
```
fun getSomeValue() {
val source1 = ...
val source2 = ...
val result = if condition(source1) source1 else source2 
...
}
```
vs
```
fun getSomeValue() = source1.run {  // the whole lambda expression intuitively associates with the source 1
  val source2 = ...
  val result = this.takeIf(source1) ?: source2 // source 2 looks incidental
}
```

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt (hunk 1 above)
Comment (epamrsa) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:60]: Moved to agent lib, so now it's possible to use just val w/o atomics.

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt (hunk 2 above)
Comment (epamrsa) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:92]: Fixed

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt (hunk 3 above)
Comment (epamrsa) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:91]: Refactored to be crossplatform.

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt (hunk 4 above)
Comment (epamrsa) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:93]: Completely refactored and moved to agent lib.

**Pull request merged**